src/ngraph_encapsulate_op.cc

```diff
@@ -262,11 +287,16 @@ class NGraphEncapsulateOp : public OpKernel {
     //
     // TODO(amprocte): Investigate performance of the compilation cache.
     if (it == m_ng_functions.end()) {
+      // Measure the total memory here first
+      mem_usage(vm0, rss0);
+
       NGRAPH_VLOG(1) << "Compilation cache miss: " << ctx->op_kernel().name();
       OP_REQUIRES_OK(
           ctx, Builder::TranslateGraph(input_shapes, static_input_map, &m_graph,
                                        ng_function));
 
+      auto function_size = ng_function->get_graph_size() / 1024.0;  // kb unit
```

Comment:
So - it seems that the function size is pretty useless at this point as it really points to constants that (if) are part of that function parameter. On the other hand - the memory used by the process is really the useful one as that reflects the correct amount of memory consumed by this process - majority of which is from the tensors they require. We will have to think through this more. Fine for now.

Reply:
Leave as is for now